src/debugger/Debugger.cpp

```diff
@@ -267,7 +267,10 @@ bool Debugger::doEval(ExecutionState* state, ByteCodeBlock* byteCodeBlock, uint8
         Value result(state->context()->globalObject()->evalLocal(*state, asValue, Value(Value::Undefined), byteCodeBlock->m_codeBlock, true));
         str = result.toStringWithoutException(*state);
     } catch (const Value& val) {
-        type = ESCARGOT_MESSAGE_EVAL_FAILED_8BIT;
+        if (!(ReferenceErrorObject*)(val.asObject())) {
+            type = ESCARGOT_MESSAGE_EVAL_FAILED_8BIT;
```

Comment:
I think that this check code seems wrong here.
when an exception occurs,  `Error` object would be thrown, and there are a number of Error types as below.
```
EvalError
InternalError
RangeError
ReferenceError
SyntaxError
TypeError
URIError
```
But in this case, you only consider `ReferenceError`, and this check code just tests if thrown value is a valid `Object`.
Since ErrorObject is one of Objects, `val.asObject()` would return the address of ErrorObject but it does not guarentee that this ErrorObject is `ReferenceError`.